README.md

```diff
@@ -33,47 +33,106 @@ workflows4regtests/basic_examples/sorting_list_fixedenv
+
+### Data and Scripts locations
+
+There is a default place where `TestKraken` tries to find all the data files and all the scripts files - this is the root directory of the tested workflow. However, in the `testkraken_spec.yml` these locations might be changed.
 
+#### `data` element
+In order to specify how to get the data, the `data` entry has to have two keys - `type` and `location`. For now, only one `type` is implemented - `workflow_path`, but in the future this might be used to specify external repositories. For `type=workflow_path`, the location is simply the relative directory path to the workflow path. An example can look like this:
+
+```yaml
+data:
+  type: workflow_path
+  location: my_data
+```
+
+#### `scripts` element
+The `scripts` entry requires only the relative directory path to the workflow path. An example can look like this:
+
+```yaml
+scripts:
+  my_scripts
+```
+Example that uses the concept can be found [here](https://github.com/ReproNim/testkraken/blob/master/workflows4regtests/basic_examples/pseudo_random_numbers/testkraken_spec.yml)
+
+### Analysis part
+The `analysis` element contains all the information required to run the workflow with the analysis. There is one required element - `command`, and two optional elements - `script` and `inputs`. The `command` could be "python" or "bash", and than the `script` will be needed, but it can be just "bet" and than the script is not required. The `inputs` part contains all the input needed to complete the command required to run the analysis. Each element of the `inputs` entry can have `type`, `argstr` (if a flag is needed) and `value`. If `type` is `File`, the file is assumed to be in the data directory. if `script` is provided, the file is expected to be in the scripts directory. An example can look like this:
+
+```yaml
 # The analysis part: inputs to the analysis script,
 # the command to run the script and the script with the analysis.
 analysis:
   inputs:
   - {type: File, argstr: -f, value: list2sort.json}
   command: python
   script: sorting.py
+```
 
+### Tests part
+The `tests` part contains all information regarding testing the analysis output. It is assumed that the output file is compared to the reference file that is available in the data directory (with the same name). There could be multiple entries for `tests`, but each element has to contain `file` with the name of the output file, `name` with the user defined name of the test, and `script` with the name of the script that should be used for running the test. The script can be saved in the script directory (checked first) or it can be from the `TestKraken` [testing_functions directories](https://github.com/ReproNim/testkraken/tree/master/testkraken/testing_functions) (the user provided tests have to follow the same template as the tests from `TestKraken` and define command line interface).
```

Comment:
```suggestion
The `tests` part contains all information regarding testing the analysis output. It is assumed that the output file is compared to the reference file that is available in the data directory (with the same name). There could be multiple entries for `tests`, but each element has to contain `file` with the name of the output file, `name` with the user defined name of the test, and `script` with the name of the script that should be used for running the test. The script can be saved in the script directory (checked first) or it can be an existing test from the `TestKraken` [testing_functions directory](https://github.com/ReproNim/testkraken/tree/master/testkraken/testing_functions). Any user provided tests have to follow the same template as the tests from `TestKraken` and define a command line interface.
```